libs/ui/label/brain/src/lib/brn-label.directive.ts

```diff
@@ -0,0 +1,43 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Directive, ElementRef, inject, Input, OnInit, PLATFORM_ID, signal } from '@angular/core';
+
+let nextId = 0;
+
+@Directive({
+	selector: '[brnLabel]',
+	standalone: true,
+	host: {
+		'[id]': '_id()',
+	},
+})
+export class BrnLabelDirective implements OnInit {
+	protected readonly _id = signal(`brn-label-${nextId++}`);
```

Comment:
Question: Is this ID stable?
- On the first page we visit, we get `brn-label-1`, then `brn-label-2`, and so on, but if I visit a new page, I would get `brn-label-3` instead of 1 again.
- If I refresh the second page, we would get `brn-label-1` instead of `brn-label-3`, since I imagine the `id` is not maintained on a per page/component basis.

This may be a non-issue in the SSG case, where pages will be pre-rendered in a particular order, so the second page above will always have `brn-label-3`, not 1. I am thinking about the SSR scenario, which I am not too familiar with.

I'm not sure what the headless-ui team does differently with React or if they have the same problem:

https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/field/field.tsx#L31-L51

https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/hooks/use-id.ts

Reply:
Yes in this case the Id will not be stable between page refreshes necessarily. I think the goal with auto generated id is to ensure that the id is absolutely unique on the page. This is good internally for us to use for ada attributes usually. If you need a persisted Id then you would need to supply it and that would be maintained always in that case. Material does id generation in the same way, check out https://github.com/angular/components/blob/main/src/material/input/input.ts but I agree with you that it does seem a little quirky that between refreshes the id continues to increment but I believe it should be a none issue

Reply:
That makes sense. Thanks for explaining!